python/pyarrow/tests/test_array.py

```diff
@@ -536,6 +536,26 @@ def test_array_slice_negative_step():
         assert result.equals(expected)
 
 
+def test_arange():
+    cases = [
+        (0, 10),  # Default step
+        (0, 10, 1),
+        (0, 10, 2),
+        (10, 0, -1),
+        (10, 0, -2),
+        (0, 0),       # Empty array
+        (0, 10, -1),  # Empty array
+        (10, 0, 1),   # Empty array
```

Comment:
These are all cases where `step` is conveniently a divisor of the length, and where either `start` or `stop` is zero. Can you make these a bit more interesting and diverse?

Reply:
I tried to improve the tests scenarios with more interesting cases